File: api/handlers/newapi.go
package handlers

import (
	"net/http"

	"github.com/drama-generator/backend/pkg/ai/newapi"
	"github.com/gin-gonic/gin"
)

// NewAPIHandler NewAPI统一接口处理器
type NewAPIHandler struct {
	client *newapi.NewAPIClient
}

// NewNewAPIHandler 创建处理器
func NewNewAPIHandler(client *newapi.NewAPIClient) *NewAPIHandler {
	return &NewAPIHandler{client: client}
}

// GenerateTextRequest 文本生成请求
type GenerateTextRequest struct {
	Model       string `json:"model" binding:"required"`
	Prompt      string `json:"prompt" binding:"required"`
	SystemPrompt string `json:"system_prompt"`
}

// GenerateImageRequest 图像生成请求
type GenerateImageRequest struct {
	Model  string `json:"model" binding:"required"`
	Prompt string `json:"prompt" binding:"required"`
	Size   string `json:"size"`
	N      int    `json:"n"`
}

// GenerateText 统一文本生成
// @Summary 统一文本生成接口
// @Description 通过NewAPI统一接口生成文本
// @Tags NewAPI
// @Accept json
// @Produce json
// @Param request body GenerateTextRequest true "请求参数"
// @Success 200 {object} newapi.AIResponse
// @Router /api/v1/newapi/text [post]
func (h *NewAPIHandler) GenerateText(c *gin.Context) {
	var req GenerateTextRequest
	if err := c.ShouldBindJSON(&req); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
		return
	}

	aiReq := &newapi.AIRequest{
		Model:        req.Model,
		Prompt:       req.Prompt,
		SystemPrompt: req.SystemPrompt,
	}

	resp, err := h.client.GenerateText(c.Request.Context(), aiReq)
	if err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
		return
	}

	c.JSON(http.StatusOK, resp)
}

// GenerateImage 统一图像生成
// @Summary 统一图像生成接口
// @Description 通过NewAPI统一接口生成图像
// @Tags NewAPI
// @Accept json
// @Produce json
// @Param request body GenerateImageRequest true "请求参数"
// @Success 200 {object} newapi.AIResponse
// @Router /api/v1/newapi/image [post]
func (h *NewAPIHandler) GenerateImage(c *gin.Context) {
	var req GenerateImageRequest
	if err := c.ShouldBindJSON(&req); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
		return
	}

	aiReq := &newapi.AIRequest{
		Model:  req.Model,
		Prompt: req.Prompt,
	}

	resp, err := h.client.GenerateImage(c.Request.Context(), aiReq)
	if err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
		return
	}

	c.JSON(http.StatusOK, resp)
}

// GetStats 获取服务商统计
// @Summary 获取统计信息
// @Description 获取各服务商的调用统计
// @Tags NewAPI
// @Produce json
// @Success 200 {object} map[string]newapi.ProviderStats
// @Router /api/v1/newapi/stats [get]
func (h *NewAPIHandler) GetStats(c *gin.Context) {
	stats := h.client.LoadBalancer.GetStats()
	c.JSON(http.StatusOK, gin.H{"stats": stats})
}

// UpdateConfig 更新配置
// @Summary 更新NewAPI配置
// @Description 更新负载均衡配置
// @Tags NewAPI
// @Accept json
// @Produce json
// @Param request body newapi.NewAPIConfig true "配置"
// @Success 200 {object} gin.H
// @Router /api/v1/newapi/config [put]
func (h *NewAPIHandler) UpdateConfig(c *gin.Context) {
	var config newapi.NewAPIConfig
	if err := c.ShouldBindJSON(&config); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
		return
	}

	h.client.BaseURL = config.BaseURL
	h.client.APIKey = config.APIKey
	h.client.SetProviders(config.Providers)

	c.JSON(http.StatusOK, gin.H{"message": "config updated"})
}

File: web/src/api/newapi.ts
import request from '@/utils/request'

// NewAPI 配置相关 API

export interface NewAPIProvider {
  name: string
  base_url: string
  api_key: string
  models: string[]
  priority: number
  weight: number
  enabled: boolean
}

export interface NewAPILoadBalancer {
  strategy: string
  providers: NewAPIProvider[]
}

export interface NewAPIConfig {
  base_url: string
  api_key: string
  load_balancer: NewAPILoadBalancer
}

export interface ProviderStats {
  name: string
  request_count: number
  error_count: number
  success_rate: number
  avg_latency: number
  last_used: string
}

/**
 * 获取NewAPI配置
 */
export function getNewAPIConfig() {
  return request({
    url: '/api/v1/newapi/config',
    method: 'get'
  })
}

/**
 * 更新NewAPI配置
 */
export function updateNewAPIConfig(data: NewAPIConfig) {
  return request({
    url: '/api/v1/newapi/config',
    method: 'put',
    data
  })
}

/**
 * 获取服务商统计
 */
export function getNewAPIStats() {
  return request({
    url: '/api/v1/newapi/stats',
    method: 'get'
  })
}

/**
 * 测试文本生成
 */
export function testTextGeneration(data: {
  model: string
  prompt: string
  system_prompt?: string
}) {
  return request({
    url: '/api/v1/newapi/text',
    method: 'post',
    data
  })
}

/**
 * 测试图像生成
 */
export function testImageGeneration(data: {
  model: string
  prompt: string
}) {
  return request({
    url: '/api/v1/newapi/image',
    method: 'post',
    data
  })
}
